terraform/hub/pod-identity.tf

```diff
@@ -0,0 +1,127 @@
+data "aws_iam_policy_document" "eks_assume" {
+  statement {
+    effect = "Allow"
+    principals {
+      type        = "Service"
+      identifiers = ["pods.eks.amazonaws.com"]
+    }
+    actions = ["sts:AssumeRole", "sts:TagSession"]
+  }
+}
+
+data "aws_iam_policy_document" "aws_assume_policy" {
+  statement {
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["sts:AssumeRole", "sts:TagSession"]
+  }
+}
+
+resource "aws_iam_policy" "aws_assume_policy" {
+  name        = "${module.eks.cluster_name}-pod-identity-aws-assume"
+  description = "IAM Policy for Pod Identity"
+  policy      = data.aws_iam_policy_document.aws_assume_policy.json
+  tags        = local.tags
+}
+
+################################################################################
+# ArgoCD EKS Access
+################################################################################
+
+resource "aws_iam_role" "argocd_hub" {
+  name               = "${module.eks.cluster_name}-argocd-hub"
+  assume_role_policy = data.aws_iam_policy_document.eks_assume.json
+}
+
+resource "aws_iam_role_policy_attachment" "argo_aws_assume_policy" {
+  role       = aws_iam_role.argocd_hub.name
+  policy_arn = aws_iam_policy.aws_assume_policy.arn
+}
+resource "aws_eks_pod_identity_association" "argocd_app_controller" {
+  cluster_name    = module.eks.cluster_name
+  namespace       = "argocd"
+  service_account = "argocd-application-controller"
+  role_arn        = aws_iam_role.argocd_hub.arn
+}
+resource "aws_eks_pod_identity_association" "argocd_api_server" {
+  cluster_name    = module.eks.cluster_name
+  namespace       = "argocd"
+  service_account = "argocd-server"
+  role_arn        = aws_iam_role.argocd_hub.arn
+}
+
+
+
+################################################################################
+# ESO EKS Access
```

Comment:
What is ESO ?

Reply:
External Secrets Operator (https://external-secrets.io/latest/) ESO :)